### PR TITLE
FEATURE: Creation of Sobel4X filtr with test, update documentation.

### PR DESCRIPTION
**Pull request by agarak**
Prepared Sobel4X filtr, added two matrix for 45 degrees (class Sobel45) and 135 degrees(class Sobel135).  Test has been written to check  every matrix.

### CODE REVIEW

File: jis-documentation/src/filters.adoc [hunk 1]
```diff
@@ -16,239 +16,151 @@ include::{sourcedir}/filters/SaltAndPepperFilterTest.java[tags=saltAndPepperFilt
 --
 == Filters
 
---
```
Comment (jgprogramming) [inline jis-documentation/src/filters.adoc]: why did you delete all of this?

File: jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/Sobel4XFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import pl.edu.uj.JImageStream.filters.edge.SobelYFilter;
+import pl.edu.uj.JImageStream.filters.edge.Sobel45Filter;
+import pl.edu.uj.JImageStream.filters.edge.Sobel135Filter;
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import pl.edu.uj.JImageStream.tests.AbstractBaseTest;
+
+public class Sobel4XFilterTest extends AbstractBaseTest {
```
Comment (jgprogramming) [inline jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/Sobel4XFilterTest.java]: this test lacks tags for code

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/edge/EdgeDetectionFilter.java [hunk 3]
```diff
@@ -8,25 +8,35 @@
...
-        int red = (int) Math.sqrt(pixelX[0] * pixelX[0] + pixelY[0] * pixelY[0]);
-        int green = (int) Math.sqrt(pixelX[1] * pixelX[1] + pixelY[1] * pixelY[1]);
-        int blue = (int) Math.sqrt(pixelX[2] * pixelX[2] + pixelY[2] * pixelY[2]);
+        int red = (int) this.dotProduct(pixelX[0], pixelY[0]);
+        int green = (int) this.dotProduct(pixelX[1], pixelY[1]);
+        int blue = (int) this.dotProduct(pixelX[2], pixelY[2]);
         int alpha = getPixel(x, y).getAlpha();
 
+
+//        int red = (int) Math.sqrt(pixelX[0] * pixelX[0] + pixelY[0] * pixelY[0]);
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/edge/EdgeDetectionFilter.java]: delete comments if they arent needed

File: jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/Sobel4XFilterTest.java (hunk 2 above)
Comment (agarak) [inline jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/Sobel4XFilterTest.java]: They have been added

**Comment by agarak** [comment]
Should I correct anything else?

File: jis-documentation/src/filters.adoc [hunk 4]
```diff
@@ -252,3 +252,26 @@ include::{sourcedir}/filters/SobelFilterTest.java[tags=edgeDetectionSobelFilter]
...
+
+===== 135 degrees
+--
+image:GrayScaleFilter.png[width="300", height="200"]
+image:Sobel135.png[width="300", height="200"]
+--
+
+===== Edge detection filter
+--
+image:Sobel.png[width="300", height="200"]
```
Comment (jgprogramming) [inline jis-documentation/src/filters.adoc]: duplicate

File: jis-documentation/src/filters.adoc [hunk 5]
```diff
@@ -252,3 +252,26 @@ include::{sourcedir}/filters/SobelFilterTest.java[tags=edgeDetectionSobelFilter]
 --
 image::Sobel.png[title="Sobel Filter", width="300", height="300"]
 --
+
+===== 45 degrees
```
Comment (jgprogramming) [inline jis-documentation/src/filters.adoc]: no titles and no code examples, like

 ```
===== Y Axis
 
 [source,java]
 .code
 ----
 include::{sourcedir}/filters/SobelFilterTest.java[tags=sobelYFilter]
 ----
 image::SobelY.png[title="SobelY Filter", width="300", height="300"]
```

**Comment by jgprogramming** [comment]
moved https://github.com/kmisztal/JImageStream/pull/53
